superset/views/core.py

```diff
@@ -1330,11 +1335,32 @@ def testconn(self):
             with closing(engine.connect()) as conn:
                 conn.scalar(select([1]))
                 return json_success('"OK"')
-        except Exception as e:
-            logger.exception(e)
+        except NoSuchModuleError as e:
+            logger.info("Invalid driver %s", e)
+            driver_name = make_url(uri).drivername
             return json_error_response(
-                "Connection failed!\n\n" f"The error message returned was:\n{e}", 400
+                _(
+                    "Could not load database driver: %(driver_name)s",
+                    driver_name=driver_name,
+                ),
+                400,
             )
+        except ArgumentError as e:
+            logger.info("Invalid URI %s", e)
+            return json_error_response(
+                _(
+                    "Invalid connection string, a valid string usually follows: \n"
+                    "'DRIVER://USER:PASSWORD@DB-HOST/DATABASE-NAME'"
```

Comment:
This got me thinking, how about moving this example connstring to `BaseEngineContext` so engine specs can override it if they feel like it?

Reply:
Will look into that, need to address tests also. This is still draft

Reply:
Not super comfortable with `BaseEngineSpec` but, If a user does not provide a valid URI we can't infer the engine, so I think it's not a good place for that error message. Note this will eventually be migrated, and then we can create some nice custom Exceptions for the API to use. 
Is this valid reasoning?

Reply:
Lol, you are absolutely right 🤦‍♂ carry on 😆